### PR TITLE
HDDS-7851. Apply flush and compact on all column families

### PR DESCRIPTION
**Pull request by symious**
## What changes were proposed in this pull request?

The operation of flush and compact only works for "default" column family.

This ticket is to also apply the operations on other column families.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-7851

## How was this patch tested?

unit test.

### CODE REVIEW

**Comment by symious** [comment]
@ChenSammi Could you help to review the PR?

**Comment by kerneltime** [comment]
Should the column family be flushed when flushing the wal as well? cc @errose28 @duongkame

**Comment by kerneltime** [comment]
Thank you at @symious for fixing this issue.

**Comment by symious** [comment]
> Should the column family be flushed when flushing the wal as well?

@kerneltime Thank you for the review. Are your refering to the function of "db.flushWal(bool)"? I think the wal is shared by all column families, should be no need of this operation.

**Comment by kerneltime** [comment]
@GeorgeJahad can you please take a look?

**Comment by duongkame** [comment]
> > Should the column family be flushed when flushing the wal as well?
> 
> @kerneltime Thank you for the review. Are your refering to the function of "db.flushWal(bool)"? I think the wal is shared by all column families, should be no need of this operation.

I think WAL is flushed (to OS cache) automatically after each writes to achieve durability (unless Ozone has a config to disable it).

On the other hand, this PR deals with flushing column families' memtables to SST files to persist the compaction after a snapshot is taken. We already flush memtable of the default column family so it makes sense to flush all the column families as well.

**Comment by sadanand48** [comment]
Some of these methods are [already implemented](https://github.com/apache/ozone/blob/HDDS-6517-Snapshot/hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RocksDatabase.java) in the snapshot branch (HDDS-6517). Please take a look.

**Review by duongkame — APPROVED** [review]
overall LGTM.

File: pom.xml [hunk 1]
```diff
@@ -1660,6 +1660,7 @@ xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xs
                       <allowedImport>org.rocksdb.StatsLevel</allowedImport>
                       <allowedImport>org.rocksdb.TransactionLogIterator.BatchResult</allowedImport>
                       <allowedImport>org.rocksdb.TickerType</allowedImport>
+                      <allowedImport>org.rocksdb.LiveFileMetaData</allowedImport>
```
[the remark this answers is not in the document]
Comment (symious) [reply pom.xml]: @duongkame Thank you for the review, removed the unused entry, please have a look.

**Comment by symious** [comment]
> Some of these methods are [already implemented](https://github.com/apache/ozone/blob/HDDS-6517-Snapshot/hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RocksDatabase.java) in the snapshot branch ([HDDS-6517](https://issues.apache.org/jira/browse/HDDS-6517)). Please take a look.

@sadanand48 Thank you for the review. I think it should be easy to resolve the conflicts, right?

**Comment by sadanand48** [comment]
> Thank you for the review. I think it should be easy to resolve the conflicts, right?

The snapshot branch got merged to master today, The conflicts need to be resolved from this PR now.

**Review by kerneltime — APPROVED** [review]

**Pull request merged**